doc/user_guide/big_data.rst

```diff
@@ -0,0 +1,227 @@
+achieved by building a "history tree" (formally called a Directed Acyclic Graph
+(DAG)) of the computations, where the original data is at the root, and any
+further operations branch from it. Only when a certain branch result is
+requested, the way to the root is found and evaluated in the correct sequence
+on the correct blocks.
+
+The "magic" is performed by (for the sake of simplicity) storing the data not
+as ``numpy.ndarray``, but ``dask.array.Array`` (more information `here
+<https://dask.readthedocs.io/en/latest/>`_). ``dask`` offers a couple of
+advantages:
+
+* **Arbitrary-sized data processing is possible**. By only loading a couple of
+  chunks at a time, theoretically any signal can be processed, albeit slower.
+  In practice, this may be limited: (i) some operations may require certain
+  chunking pattern, which may still saturate memory; (ii) many chunks should
+  fit into the computer memory comfortably at the same time.
+* **Loading only the required data**. If a certain part (chunk) of the data is
+  not required for the final result, it will not be loaded at all, saving time
+  and resources.
+* **Able to extend to a distributed computing environment (clusters)**.
+  ``dask.distributed`` (documentation `here
+  <https://distributed.readthedocs.io/en/latest/>`_) offers a straightforward
+  way to expand the effective memory for computations to that of a cluster,
+  which allows performing the operations significantly faster than on a single
+  machine.
+
+
+Creating Lazy Signals
+---------------------
+
+Loading lazily
+^^^^^^^^^^^^^^
+
+To load the data lazily, pass the keyword ``lazy=True``. Note that this
+deprecates and replaces all ``memmap`` and ``load_to_memory=False`` and similar
+options. As an example, loading a 34.9 GB ``.blo`` file on a regular laptop
+might look like:
+
+.. code-block:: python
+
+    >>> s = hs.load("shish26.02-6.blo", lazy=True)
+    >>> s
+    <LazySignal2D, title: , dimensions: (400, 333|512, 512)>
+    >>> s.data
+    dask.array<array-e..., shape=(333, 400, 512, 512), dtype=uint8, chunksize=(20, 12, 512, 512)>
+    >>> print(s.data.dtype, s.data.nbytes / 1e9)
+    uint8 34.9175808
+    >>> s.change_dtype("float") # To be able to perform decomposition, etc.
+    >>> print(s.data.dtype, s.data.nbytes / 1e9)
+    float64 279.3406464
+
+Loading the dataset in the original unsigned integer format would require
+around 35GB of memory. To store it in a floating-point format one would need
+almost 280GB of memory. However, with the lazy processing both of these steps
+are near-instantaneous and require very little computational resources.
+
+Lazy stacking
+^^^^^^^^^^^^^
+
+Occasionally the full dataset consists of many smaller files. To combine them
+into a one large ``LazySignal``, we can :ref:`stack<signal.stack_split>` them
+lazily (both when loading of afterwards):
+
+.. code-block:: python
+
+    >>> siglist = hs.load("*.hdf5")
```

Comment:
Yes. Maybe I failed to illustrate what I wanted though - the goal was to show that a lazy signal can be constructed by lazily stacking many normal signals (e.g. tiff images, that may have been processed separately and already open, or something).

Admittedly, the only example I can think of where you may want lazy stack but normal signals is if you only have enough memory for one full copy (e.g. it takes up 60% of your memory), and want to be able to compare the original files with the processed stack. Though I would load the individual files lazily any way... Maybe if it's not straightforward (something similar to bcf)?